src/org/musicbrainz/webservice/impl/HttpClientWebServiceWs2.java

```diff
@@ -185,19 +179,12 @@ protected Metadata doPost(String url, Metadata md) throws WebServiceException, M
         HttpClient httpClient = setConnectionParam(true);
 
         HttpPost method = new HttpPost(url);
-  
-        try {
 
-            StringEntity httpentity = new StringEntity(getWriter().getXmlString(md));
-            httpentity.setContentType(new BasicHeader("Content-Type", "application/xml; charset=UTF-8"));
+        StringEntity httpentity = new StringEntity(getWriter().getXmlString(md), Consts.UTF_8);

```

Comment:
Same here: Let's keep the changes to a bare minimum. While I agree that the error handling here might not be the best idea, changing it might cause problems we're not aware of and break backward compatibility.

Or did the error handling cause problems for you?

Reply:
From what I see the catch block was only there because the default constructor was able to throw the checked exception UnsupportedEncodingException which is not the case if you pass the encoding directly. Using the new constructor no IOExceptions can be thrown from the block wrapped by try.